modules/auxiliary/scanner/http/influxdb_enum.rb

```diff
@@ -0,0 +1,78 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'InfluxDB Enum Utility',
+      'Description'    => %q{
+        This module enumerates databases on InfluxDB using the REST API
+        (using default authentication - root:root).
+      },
+      'References'     =>
+        [
+          ['URL', 'http://influxdb.com/docs/v0.9/concepts/reading_and_writing_data.html']
+        ],
+      'Author'         => [ 'Roberto Soares Espreto <robertoespreto[at]gmail.com>' ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(8086),
+        OptString.new('TARGETURI', [true, 'Path to list all the databases', '/db']),
+        OptString.new('USERNAME', [true, 'The username to login as', 'root']),
+        OptString.new('PASSWORD', [true, 'The password to login with', 'root'])
+      ], self.class)
+  end
+
+  def run
+    username = datastore['USERNAME']
+    password = datastore['PASSWORD']
+
+    res = send_request_cgi(
+      'uri'           => normalize_uri(target_uri.path),
+      'method'        => 'GET',
+      'authorization' => basic_auth(username, password)
```

Comment:
send_request_cgi is actually pretty smart about this so you don't have to explicitly handle basic auth for your HTTP request, it should automatically do it for you. Here's the code that does that:
https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/proto/http/client.rb#L217

So if I'm not mistaken, you can simply remove this line, and it should auth anyway (if it detects a 401 WWW-Authenticate).